### PR TITLE
Menu Restructure

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
Add-on menu restructure changes.
These are related to the eferenced PR, but should be safe to merge independantluy of that.

## Related Issues
* https://github.com/zaproxy/zaproxy/pull/8358
* https://github.com/zaproxy/zaproxy/issues/8369

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by psiinon** [comment]
I've not updated any changelogs as I'm expecting comments and dont want to have to do a load of conflict resolution 😉 
So plan to add them when it looks ready to merge..

**Comment by kingthorin** [comment]
Is the plan to remove the commonlib piece after the next core release?

**Comment by psiinon** [comment]
I was thinking we could change it to extend the core class, remove the existing statics, and then add any new ones as needed.
That way the other ext classes won't need to change

**Comment by psiinon** [comment]
All changes done (inc the scripts panel menus) so this is ready for review..

File: addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties
Comment (kingthorin) [inline addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:1]: Thanks for this, the length of some of these had been bugging me

**Review by kingthorin — APPROVED** [review]

File: addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties
Comment (psiinon) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:1]: Me too 😉

**Comment by kingthorin** [comment]
One of the build files is missing a closing quote.

**Comment by psiinon** [comment]
ARG - broke it, need to repush

**Comment by psiinon** [comment]
Should be good one file should have been removed. GitHub looks like its struggling to changed to WIP until its back in sync

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties [hunk 1]
```diff
@@ -56,14 +56,14 @@ exim.importurls.warn.scheme = "{0}" does not have a scheme.
 
 exim.menu.copyurls.popup = Copy URLs
 exim.menu.export = Export
-exim.menu.export.context.urls = Export URLs for Context(s)
-exim.menu.export.messages.popup = Export Messages to File...
+exim.menu.export.context.urls = Save URLs...
```
Comment (ricekot) [inline addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: It seems the changes here led to duplicate menu items under Export,

Before
image

After
image

Comment (thc202) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: Given that's already in the context menu I'd drop it.

Comment (ricekot) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: These are automatically removed from the Export menu and shown in the context menu when there are URLs in the Sites Tree. I guess that's another bug (the same component cannot be in two menus?)

I'm okay with dropping them, but we should be consistent. The other items under Export can also move into relevant context menus.

For now I would prefer duplicating the PopupMenuItem component with different names and adding to both the context menu as well as the Export menu (e.g. `Save URLs` and `Save Context URLs`).

Comment (thc202) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: That sounds like a core bug.

I'm fine with that but don't really see the advantage of duplicating them when they are already available in the correct context menu.

Comment (kingthorin) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: There's also a duplication issue in the Site's Tree (and maybe history?) context menu, under "Include in Context" there is now always two "New Context" entries.

Comment (ricekot) [reply addOns/exim/src/main/resources/org/zaproxy/addon/exim/resources/Messages.properties:59]: See also #6012.

**Pull request merged**